src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -95,37 +95,52 @@ internal partial class CodeFixService : ForegroundThreadAffinitizedObject, ICode
             }
 
             using var _ = ArrayBuilder<DiagnosticData>.GetInstance(out var diagnostics);
-            using var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
-
-            var linkedToken = linkedTokenSource.Token;
 
-            // This flag is used by SuggestedActionsSource to track what solution is was
-            // last able to get "full results" for.
+            // This flag is used by SuggestedActionsSource to track what solution is was last able to get "full results" // for.
             var isFullResult = await _diagnosticService.TryAppendDiagnosticsForSpanAsync(
-                document, range, diagnostics, cancellationToken: linkedToken).ConfigureAwait(false);
+                document, range, diagnostics, cancellationToken: cancellationToken).ConfigureAwait(false);
 
             var errorDiagnostics = diagnostics.Where(d => d.Severity == DiagnosticSeverity.Error);
             var otherDiagnostics = diagnostics.Where(d => d.Severity != DiagnosticSeverity.Error);
 
-            // Kick off a task that will determine there's an Error Diagnostic with a fixer
-            var errorDiagnosticsTask = Task.Run(
-                () => GetFirstDiagnosticWithFixAsync(document, errorDiagnostics, range, linkedToken),
-                linkedToken);
-
-            // Kick off a task that will determine if any non-Error Diagnostic has a fixer
-            var otherDiagnosticsTask = Task.Run(
-                () => GetFirstDiagnosticWithFixAsync(document, otherDiagnostics, range, linkedToken),
-                linkedToken);
-
-            // If the error diagnostics task happens to complete with a non-null result before
-            // the other diagnostics task, we can cancel the other task.
-            var diagnostic = await errorDiagnosticsTask.ConfigureAwait(false)
-                ?? await otherDiagnosticsTask.ConfigureAwait(false);
-            linkedTokenSource.Cancel();
-
-            return new FirstDiagnosticResult(partialResult: !isFullResult,
-                                   hasFix: diagnostic != null,
-                                   diagnostic: diagnostic);
+            // Kick off a task that will determine there's an Error Diagnostic with a fixer.
+            var errorDiagnosticTask = Task.Run(
+                () => GetFirstDiagnosticWithFixAsync(document, errorDiagnostics, range, cancellationToken),
+                cancellationToken);
```

Comment:
intentional switch to cancellationToken from linkedToken.  this code is awaited prior to cancelling the linked token.  so using the linked token serves no purpose.